rzup/src/bin/rzup.rs

```diff
@@ -0,0 +1,157 @@
+                version: opts.version,
+            }
+            .run()
+            .expect("Error during cargo-risczero installation");
+            InstallToolchain {
+                toolchain: Some("latest".to_string()),
+                repo: ToolchainRepo::Rust,
+            }
+            .run()
+            .expect("Error during Rust toolchain installation");
+        }
+        RzupSubcmd::Show { verbose, subcmd } => handle_show(verbose, subcmd),
+        RzupSubcmd::Update { .. } => handle_update(),
+        RzupSubcmd::Check => handle_check_all().expect("Error checking for updates"),
+        RzupSubcmd::Default { opts } => handle_default(opts),
+        RzupSubcmd::Toolchain { subcmd } => handle_toolchain(subcmd),
+        RzupSubcmd::Extension { subcmd } => handle_extension(subcmd),
+        RzupSubcmd::Self_ { .. } => {
+            println!("Self not yet implemented")
+        }
+    }
+}
+
+#[derive(Debug, Parser)]
+#[command(
+    name = "rzup",
+    bin_name = "rzup[EXE]",
+    version = utils::version(),
+    before_help = format!(
+"rzup {}
+
+    ██████  █▀▀▀██
+    ██████  █ ▄▀ █        Prove anything.
+    ██████  ██▄▄▄█       Verify everywhere.
+    ██████
+    ██████   RISC
+    ██████   ZERO
+
+Contribute      : https://github.com/risc0/
+Documentation   : https://dev.risczero.com/
+Chat & Support  : https://discord.com/invite/risczero/",
+    utils::version()
+    ),
+    after_help = help::RZUP_HELP
+)]
+pub struct Rzup {
+    /// Enable verbose output
+    #[arg(short, long)]
+    verbose: bool,
+
+    #[command(subcommand)]
+    pub subcmd: Option<RzupSubcmd>,
+}
+
+#[derive(Debug, Subcommand)]
+#[command(name = "rzup", bin_name = "rzup[EXE]")]
+pub enum RzupSubcmd {
+    #[command(after_help = help::INSTALL_HELP, hide = true)]
+    Install {
+        #[command(flatten)]
+        opts: InstallOpts,
+    },
+    /// Show the active and installed RISC Zero toolchains
+    #[command(after_help = help::SHOW_HELP)]
+    Show {
+        /// Enable verbose output for all installed toolchains
+        #[arg(short, long)]
+        verbose: bool,
+
+        #[command(subcommand)]
+        subcmd: Option<ShowSubcmd>,
+    },
+    /// Update RISC Zero toolchains and cargo-risczero
+    #[command(after_help = help::UPDATE_HELP, aliases = ["upgrade", "up"])]
+    Update {
+        /// Toolchain name, such as latest, tag.
+        #[arg(num_args = 1..)]
+        toolchain: Vec<String>, // TODO
+
+        /// Force an update
+        #[arg(long)]
+        force: bool,
+    },
+    /// Check for updates to RISC Zero toolchains and cargo-risczero
+    Check,
+    /// Set the default RISC Zero toolchain
+    Default {
+        #[command(flatten)]
+        opts: DefaultOpts,
+    },
+    /// Manage and install RISC Zero toolchains
+    Toolchain {
+        #[command(subcommand)]
+        subcmd: ToolchainSubcmd,
+    },
+    /// Manage cargo-risczero extensions
```

Comment:
This is a bit of a confusing hint message to me. Will there be other extensions supported? Or is this just planned to update the `cargo-risczero` version?

Reply:
My goal was to make this future proof, and allow many versions of cargo-risczero, or even other lang "extensions". Though this might be better to kick this can down the road with less confusing wording.

Reply:
What is an extension?

Reply:
cargo-risczero is an example of a cargo "extension", but perhaps its just not a very useful way of describing it here if its unclear.